src/libsync/discovery.cpp

```diff
@@ -551,6 +575,8 @@ void ProcessDirectoryJob::processFileAnalyzeRemoteInfo(
             }
             item->_instruction = CSYNC_INSTRUCTION_UPDATE_METADATA;
             item->_direction = SyncFileItem::Down;
+        } else if(isServerEntryLocked != dbEntry._lockstate._locked) {
```

Comment:
@claucambra Are you sure this will not change sync behavior? Like, if the file has been also changed on the server, with this `else if`, we will never go into `else's` logic...

Reply:
@claucambra My assumption could be wrong, so, just to be more precise, what I want is for us to make sure we only need to update metadata and nothing else besides lock state change has happened with the file.

Reply:
I am not sure, so to be safe I have moved this check onto a lambda that gets called in the final and the outside branch of the if